content/chapters/compute/lab/content/arena.md

```diff
@@ -239,3 +239,57 @@ The code in `test.c` verifies its single-threaded correctness, while the one in
 Your task is to synchronize this data structure using whichever primitives you like.
 Try to keep the implementation efficient.
 Aim to decrease your running times as much as you can.
+
+## Minor and Major Page Faults
+
+The code in `support/page-faults/page_faults.c` generates some minor and major page faults.
+Open 2 terminals: one in which you will run the program, and one which will monitor the page faults of the program.
+In the monitoring terminal, run the following command:
+
+```console
+watch -n 1 'ps -eo min_flt,maj_flt,cmd | grep ./page_faults | head -n 1'
+```
+
+Compile the program and run it.
+You must press `enter` one time, before the program will prompt you to press `enter` more times.
+Watch the first number on the monitoring terminal;
+it increases.
+Those are the minor page faults.
+
+### Minor Page Faults
+
+A minor page fault is generated whenever access toa page is requested, the page is present in the physical memory, as a frame, but that frame isn't allocated to the process generating the page fault.
+These types of page faults are the most common, and they happen when calling functions from dynamic libraries, allocating heap memory, loading programs, reading files that have been cached, and many more situations.
+Now back to the program.
+
+The monitoring command already starts with some minor page faults, generated when loading the program.
+
+After pressing enter, the number increases, because a function from a dynamic library (libc) is fetched, when the first `printf` is executed.
```

Comment:
```suggestion
After pressing `enter`, the number increases, because a function from a dynamic library (libc) is fetched when the first `printf()` is executed.
```